src/factories/node.ts

```diff
@@ -14,12 +14,15 @@ export type NodeContainerFactory = Awaited<ReturnType<typeof nodeContainerFactor
 
 // eslint-disable-next-line @typescript-eslint/explicit-function-return-type
 export async function nodeContainerFactory(node: RunGraphNode) {
+  const application = await waitForApplication()
   const cull = await waitForCull()
   const { animate } = await animationFactory()
   const { element: container, render: renderNode, bar } = await getNodeFactory(node)
-  const cacheKey: string | null = null
 
+  let internalNode = node
+  let cacheKey: string | null = null
```

Comment:
Cache key was never getting set before so every render call was a cache miss. Actually setting the cacheKey now and fixed the logic for creating a cache key. Previously was caching on `end_date.toString()` which only had second precisions. The cache key is now just based on the `state_type` and `end_date.getTime()` values.